Issue 1330 documentation changes (#3243)
grafana/mimir: docs/sources/operators-guide/run-production-environment/production-tips/index.md

The per-tenant TSDB is composed of several files and the ingester keeps a file descriptor open for each TSDB file.
The total number of file descriptors, used to load TSDB files, linearly increases with the number of tenants in the Grafana Mimir cluster and the configured `-blocks-storage.tsdb.retention-period`.

> To avoid reaching the maximum number of open file descriptors, fine tune the following settings:

Reviewer: Nice. :)

### Ensure caching is enabled

> The querier supports caching. Ro reduce the number API calls to the long-term storage, enable caching in the querier.

Reviewer: ```suggestion
The querier supports caching. To reduce the number API calls to the long-term storage, enable caching in the querier.
```
Nice, even with with the typo. :)

The store-gateway keeps a file descriptor open for each index-header loaded at a given time.
The total number of file descriptors used to load index-headers linearly increases with the number of blocks owned by the store-gateway instance.

> Configure the system's `file-max` ulimit at least to `65536` to avoid reaching the maximum number of open file descriptors.

Reviewer: ```suggestion
To avoid reaching the maximum number of open file descriptors, configure the system's `file-max` ulimit to at least `65536`.
```

Reviewer: Rationale (I think, as a native speaker who sometimes does not know why) for changing the word order: "at least" is more tightly coupled / collocated with the value `65536` than it is to "ulimit". If any grammar nerds want to pipe up, I’d be curious why I changed this. :)

### Ensure Memcached is properly scaled

> Ensure Memcached evictions happen infrequently.
> Grafana Mimir query performances might be negatively affected if your Memcached cluster evicts items frequently.

Reviewer: ```suggestion
Grafana Mimir query performances might be negatively affected if your Memcached cluster evicts items frequently.
To prevent this, make sure that Memcached evictions happen infrequently.
```

> Increase your Memcached cluster replicas to add more memory to the cluster and reduce evictions.

Reviewer: ```suggestion
To add more memory to the cluster and reduce evictions, increase your Memcached cluster replicas.
```

> We also recommend running a dedicated Memcached cluster for each type of cache: query results, metadata, index, and chunks. While not required, it ensures that each cache is isolated from the others.

Reviewer: ```suggestion
To make sure that each cache is isolated from the another, (optionally) run a dedicated Memcached cluster for each type of cache: query results, metadata, index, and chunks.
```
